### ISSUE TITLE
role="separator" should not require aria-valuenow when tabIndex=-1

### ISSUE THREAD

**Issue by bvandercar-vt** [dequelabs/axe-core#4062]
### Product

axe-core

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
Should not throw `aria-required-attr` on a `role="separator"` element when that element has `tabIndex=-1` and is missing `aria-valuenow`. It fails because it says `aria-valuenow` property is required, however this property is **only applicable when the `separator` is focusable**. This is congruent with the documentation here: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/separator_role .

#### Actual
Throws `aria-required-attr` rule when `separator` is not focusable and is missing `aria-valuenow`.

**Comment by straker** [comment]
Thanks for the issue. The [spec also agrees with you](https://www.w3.org/TR/wai-aria-1.2/#separator) that they are only required if focusable. The issue is that axe-core treats `tabindex=-1` as focusable as it allows JavaScript to still focus the element. You're best bet is to remove the `tabindex` attribute and use the role on a non-interactive element.

**Comment by WilcoFiers** [comment]
Looks like this is getting resolved. Closing
